examples/agentInteractionsDemo/app.js

```diff
@@ -207,9 +209,22 @@ new Vue({
                                 .then((conv) => {
                                     let newConv = conv;
                                     newConv.customer = conv.participants.find((part) => {
-                                        return part.purpose === "customer";
-                                    });
-                                    this.conversationsData.conversations.push(newConv);
+                                        return part.purpose === "customer" || part.purpose === "external";
+                                    }) || { name: "Unknown" };
+
+                                    if (newConv.customer.externalContactId){
+                                        console.log("externalContact", newConv)
+                                        externalContactsApi.getExternalcontactsContact(newConv.customer.externalContactId)
+                                            .then((externalContact) => {
+                                                newConv.customer.name = `${externalContact.firstName} ${externalContact.lastName}`;
+                                                this.conversationsData.conversations.push(newConv);
```

Comment:
below, on 228, the code attempts to set a unique array and also builds up a map of convo id to evals.  I think the eval map will still work with this additional promise and not waiting on it, but, the unique array might be a bit wonky.

In an ideal world, I think we make an async util call that resolves with the convos and evals and then just assign it to vue's model.  That way, we're not constantly repainting when each conversation details resolve.  But, you could also just extract the uniqueness and map building into a function and call that in two places if you dont want to get into all this for a bugfix.  Dealer's choice.